onnxruntime/core/providers/qnn/builder/opbuilder/conv_op_builder.cc

```diff
@@ -209,13 +209,15 @@ Status ConvOpBuilder::ProcessConv2DInputs(QnnModelWrapper& qnn_model_wrapper,
       return ORT_MAKE_STATUS(ONNXRUNTIME, FAIL, "QNN EP: Unexpected convolution op type: ", node_unit.OpType().c_str());
     }
 
+    bool is_3d = (input_info.shape.size() == 5);
+
     std::vector<uint8_t> unpacked_tensor;
     if (input_info.is_initializer) {
       // Get transposed initializer bytes.
```

Comment:
Looks like this is not yet handling per-channel quant for this weight input. I believe we'll need to use a 3d transpose permutation in line 227:

```c++
      // Transpose quantization parameter's axis if this is using per-channel quantization.
      if (input_info.quant_param.IsPerChannel()) {
        // TODO: use 3d permutation to transpose axis.
        const std::vector<size_t>& perm = conv_type == OnnxConvType::kConv ? nchw2hwcn_perm : cnhw2hwcn_perm;
        std::vector<size_t> perm_inv(perm.size());
        ORT_RETURN_IF_ERROR(utils::InvertPerm<size_t>(perm, perm_inv));
        ORT_RETURN_IF_ERROR(input_info.quant_param.HandleTranspose<size_t>(perm_inv));
      }
```

Reply:
Can we also add a unit test for 3d per-channel conv?